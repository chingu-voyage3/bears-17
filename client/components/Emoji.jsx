import React from 'react';
import PropTypes from 'prop-types';

const Emoji = props => (
  <span className="emoji" role="img" aria-label={props.label}>
    {props.children}
  </span>
);

Emoji.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Emoji;
