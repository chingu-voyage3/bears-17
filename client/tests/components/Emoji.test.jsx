/* eslint-disable jsx-a11y/accessible-emoji */
import React from 'react';
import renderer from 'react-test-renderer';

import Emoji from 'Components/Emoji';

describe('Emoji', () => {
  it('should match the snapshot', () => {
    const emoji = renderer.create(<Emoji label="fire">ðŸ”¥</Emoji>);

    expect(emoji.toJSON()).toMatchSnapshot();
  });
});
