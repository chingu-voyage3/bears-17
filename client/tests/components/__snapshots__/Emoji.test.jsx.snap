// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji should match the snapshot 1`] = `
<span
  aria-label="fire"
  className="emoji"
  role="img"
>
  ðŸ”¥
</span>
`;
